toolkit/tools/pkg/profile/profile.go

```diff
@@ -0,0 +1,76 @@
+package profile
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"runtime/trace"
+)
+
+type Profiler struct {
+	cpuProfFile *os.File
+	memProfFile *os.File
+	traceFile   *os.File
+}
+
+func StartProfiling(cpuProfFile, memProfFile, traceFile string, enableCpuProf, enableMemProf, enableTrace bool) (*Profiler, error) {
```

Comment:
Doesn't need to be now, but adding a readme on how to consume this data would be nice as well.

Reply:
Sure, will make another PR